src/main/java/jenkins/plugins/logstash/configuration/ElasticSearch.java

```diff
@@ -16,24 +17,27 @@
 
 public class ElasticSearch extends LogstashIndexer<ElasticSearchDao>
 {
-  protected String key;
-  protected String username;
-  protected Secret password;
+  private String username;
+  private Secret password;
+  private URI uri;
 
   @DataBoundConstructor
-  public ElasticSearch()
+  public ElasticSearch(String uri) throws URISyntaxException
   {
+    this.uri = new URI(uri);
+    validateUri(this.uri);
   }
 
-  public String getKey()
+  public URI getUri()
   {
-    return key;
+    return uri;
   }
 
-  @DataBoundSetter
-  public void setKey(String key)
+  public void setUri(String value) throws URISyntaxException
```

Comment:
minor; for sake of consistency can we make setter accept URI?

Reply:
With URI stapler is not able to autoconvert the String to an URI. I will switch to URL as persistence here. Then I can also use a DataBoundSetter that accepts a URL